src/jit/ee_il_dll.cpp

```diff
@@ -163,6 +163,20 @@ CorJitResult CILJit::compileMethod (
         return g_realJitCompiler->compileMethod(compHnd, methodInfo, flags, entryAddress, nativeSizeOfCode);
     }
 
+    if (!JitConfig.isInitialized())
```

Comment:
Then it looks like there is a bug. The existing config values (which are all statics) are accessed like using `ConfigX::val`, which boils down to this:

```
inline DWORD val(const CLRConfig::ConfigDWORDInfo & info)
{
    WRAPPER_NO_CONTRACT;
    // make sure that the memory was zero initialized
    _ASSERTE(m_inited == 0 || m_inited == 1);
    if (!m_inited) init(info);
    return m_value;
}

void init(const CLRConfig::ConfigDWORDInfo & info)
{
    CONTRACTL
    {
        NOTHROW;
    }
    CONTRACTL_END;
    
    // make sure that the memory was zero initialized
    _ASSERTE(m_inited == 0 || m_inited == 1);

    m_value = CLRConfig::GetConfigValue(info);
    m_inited = 1;
}
```